EFCache.Redis/RedisCache.cs

```diff
@@ -167,75 +170,105 @@ public bool GetItem(string key, out object value)
 			}
 		}
 
+		// Note: This method does not use try/catch. It attempts retry and then throws if unsuccessful
+		// This method should never be allowed to proceed with an exception as it will lead to
+		// an incoherent cache
 		public void InvalidateSets(IEnumerable<string> entitySets)
 		{
 			// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
 			entitySets.GuardAgainstNull(nameof(entitySets));
 
 			var database = Connection.GetDatabase();
 
-			try
+			var keysAndEntitySetsToInvalidate = new HashSet<(string, RedisKey)>();
+			// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
+			foreach (var entitySet in entitySets)
 			{
-				var keysAndEntitySetsToInvalidate = new HashSet<(string, RedisKey)>();
-				// ReSharper disable once PossibleMultipleEnumeration - the guard clause should not enumerate, its just checking the reference is not null
-				foreach (var entitySet in entitySets)
-				{
-					var entitySetKey = AddCacheQualifier(entitySet);
-					var keys = database.SetMembers(entitySetKey).Select(v => (v.ToString(), entitySetKey));
-					keysAndEntitySetsToInvalidate.UnionWith(keys);
-				}
-
-				var transaction = database.CreateTransaction();
-				foreach (var (key, entitySetKey) in keysAndEntitySetsToInvalidate)
-				{
-					var hashedKey = HashKey(key);
-					transaction.KeyDeleteAsync(hashedKey);
-					transaction.SetRemoveAsync(entitySetKey, hashedKey);
-				}
-				var committed = transaction.Execute();
+				var entitySetKey = AddCacheQualifier(entitySet);
+				var keys = database.SetMembers(entitySetKey).Select(v => (v.ToString(), entitySetKey));
+				keysAndEntitySetsToInvalidate.UnionWith(keys);
+			}
 
-				if (!ShouldCollectStatistics || !committed) return;
-				
-				foreach (var (key, _) in keysAndEntitySetsToInvalidate)
-				{
-					database.HashIncrement(AddStatsQualifier(HashKey(key)), InvalidationsIdentifier, 1, CommandFlags.FireAndForget);
-				}
-				
+			var transaction = database.CreateTransaction();
+			foreach (var (key, entitySetKey) in keysAndEntitySetsToInvalidate)
+			{
+				var hashedKey = HashKey(key);
+				transaction.KeyDeleteAsync(hashedKey);
+				transaction.SetRemoveAsync(entitySetKey, hashedKey);
 			}
-			catch (Exception e)
+
+			var committed = false;
+			PerformWithRetryBackoff(() =>
 			{
-				OnCachingFailed(e);
```

Comment:
@jkdufair what are the side effects of not calling `OnCachingFailed`? or rather, when should it be called?

Reply:
See above re: not calling it. When should it be called? I guess maybe never??? Maybe we should only log to Sentry when it really fails. Transient stuff may not be as important.